File: android/app/src/main/java/com/legaltalk/india/data/model/LawyerProfile.kt
package com.legaltalk.india.data.model

data class LawyerProfile(
    val userId: String,
    val barCouncilNumber: String,
    val stateBarCouncil: String,
    val aadhaar: String,
    val pan: String,
    val bio: String,
    val experienceYears: Int,
    val languages: List<String>,
    val categories: List<String>,
    val chatPricePerMinute: Double,
    val voicePricePerMinute: Double,
    val videoPricePerMinute: Double,
    val verificationStatus: String, // "pending", "approved", "rejected", "suspended"
    val isOnline: Boolean,
    val rating: Double,
    val reviewCount: Int,
    val practiceState: String? = null,
    val practiceDistrict: String? = null,
    val llbGraduationYear: Int? = null,
    val llbUniversity: String? = null,
    val barAssociationName: String? = null,
    val placeOfPractice: String? = null,
    val enrollmentCertificateUrl: String? = null,
    val copUrl: String? = null,
    val llbCertificateUrl: String? = null,
    val subscriptionExpiresAt: String? = null,
    
    // joined fields from users association
    val name: String? = null,
    val avatarUrl: String? = null,
    val mobile: String? = null,
    val email: String? = null,
    val isBlocked: Boolean? = null
)

File: ios/LegalTalkIndia/Data/Model/LawyerProfile.swift
import Foundation

struct LawyerProfile: Codable, Identifiable {
    var id: String { userId }
    let userId: String
    let barCouncilNumber: String
    let stateBarCouncil: String
    let aadhaar: String
    let pan: String
    let bio: String
    let experienceYears: Int
    let languages: [String]
    let categories: [String]
    let chatPricePerMinute: Double
    let voicePricePerMinute: Double
    let videoPricePerMinute: Double
    let verificationStatus: String // "pending", "approved", "rejected", "suspended"
    let isOnline: Bool
    let rating: Double
    let reviewCount: Int
    let practiceState: String?
    let practiceDistrict: String?
    let llbGraduationYear: Int?
    let llbUniversity: String?
    let barAssociationName: String?
    let placeOfPractice: String?
    let enrollmentCertificateUrl: String?
    let copUrl: String?
    let llbCertificateUrl: String?
    let subscriptionExpiresAt: String?
    
    // joined fields from users table representation
    let name: String?
    let avatarUrl: String?
    let mobile: String?
    let email: String?
    let isBlocked: Bool?
}
